### PR TITLE
Add option to store unconstrained draws and update stan interface

### PR DESCRIPTION
**Pull request by aseyboldt**
- `sample(store_unconstrained=True)` will add an additional sampling stat, the current draw in unconstrained space. This can be useful for debugging.
- The stan interface was a bit outdated. We should add tests for those, this is a bit tricky though because they require a patched httpstan (mentioned in the readme)
- We also pass seed chain and draw to the sample expanding function, I think that needs to be set in the stan expansion function to get good generated quantities (that might be random...)

### CODE REVIEW

File: nutpie/sample.py [hunk 1]
```diff
@@ -27,10 +27,12 @@ def sample(
     tune: int = 1000,
     chains: int = 4,
     seed: int = 42,
-    num_try_init=100,
+    num_try_init=200,
```
Comment (ricardoV94) [inline nutpie/sample.py:30]: Wow, that's high. I think we try like 10x times only in PyMC, but we do it per variable perhaps?

File: nutpie/sample.py [hunk 2]
```diff
@@ -27,10 +27,12 @@ def sample(
...
     save_warmup: bool = True,
     store_divergences: bool = False,
     progress_bar=True,
+    init_mean=None,
+    store_unconstrained=False,
```
Comment (ricardoV94) [inline nutpie/sample.py:35]: Missing from the docstrings

File: nutpie/sample.py (hunk 1 above)
Comment (aseyboldt) [reply nutpie/sample.py:30]: I adapted this from stan a bit, they try even more often. I don't think it matters that much to be honest, but trying a few more times might make things a bit more robust, and it doesn't cost much...

Comment (ricardoV94) [reply nutpie/sample.py:30]: Yeah, I was not concerned, just surprised

File: nutpie/sample.py (hunk 2 above)
Comment (aseyboldt) [reply nutpie/sample.py:35]: fixed, I also added a couple other missing args (from the kwargs)

**Review by ricardoV94 — APPROVED** [review]

**Pull request merged**